### PR TITLE
Improve comparators

### PR DESCRIPTION
**Pull request by michamos**
This makes several changes to comparators:
* it uses `source` for comparing `titles` and `source` and `value` for `dois`
* it improves comparison of `references` by requiring some fields in the `publication_info` to be present before a match is declared. Note that this will result in more duplicate elements in the reference list after the merge due to references that contain too little information to be matched.

### CODE REVIEW

File: inspire_json_merger/comparators.py [hunk 1]
```diff
@@ -85,6 +85,28 @@ class Ret(PrimaryKeyComparator):
...
+    If any required key is not present, the element won't match.
+    """
+    def __init__(self, keys):
+        self.keys = set(keys)
+
+    def __call__(self, field):
+        if not self.keys.issubset(field.keys()):
+            return NoMatch()
+
+        return {k: field[k] for k in self.keys}
```
Comment (michamos) [inline inspire_json_merger/comparators.py]: I did something simple here. Ideally we would support something similar to the `PrimaryKeyComparator`, but that's a bit more complex to implement.

File: inspire_json_merger/comparators.py [hunk 2]
```diff
@@ -103,15 +125,19 @@ class Ret(PrimaryKeyComparator):
...
+        ['raw_refs'],
+        ['reference.arxiv_eprint'],
+        ['reference.dois'],
+        ['reference.isbn'],
+        ['reference.report_numbers'],
+        ['reference.persistent_identifiers'],
+        ['reference.publication_info']
+    ],
+    {'reference.publication_info': SubdictNormalizer(['journal_title', 'journal_volume', 'page_start'])}
+)
```
Comment (david-caro) [inline inspire_json_merger/comparators.py]: David approves this indentation :1st_place_medal:

File: tests/unit/test_merger_arxiv2arxiv.py [hunk 3]
```diff
@@ -1289,9 +1289,13 @@ def test_merging_dois_field():
...
         'dois': [
             {
-                'material': 'publication',
+                'material': 'preprint',
                 'source': 'nowhere',
                 'value': '10.1023/A:1026654312961'
+            },
+            {
+                'material': 'publication',
+                'value': '10.1023/A:1026654312961'
```
Comment (david-caro) [inline tests/unit/test_merger_arxiv2arxiv.py]: I'd add a test that does actually match the dois also, as we just lost it (there was none that did not though).

Comment (michamos) [reply tests/unit/test_merger_arxiv2arxiv.py]: :+1: actually you made me realize the new comparator is wrong when we have two DOIs from the same publisher, because of errata typically. I will need to fix that.
